backend/test/email/emailHandler.test.ts

```diff
@@ -102,5 +109,59 @@ describe("emailHandler", () => {
         emailClientStub.verifyEmailsSent();
       });
     });
+
+    describe("sendNahtavillaoloMails", () => {
+      let projekti: DBProjekti;
+      beforeEach(() => {
+        projekti = new DBProjektiForSpecificVaiheFixture().getProjektiForVaihe(Vaihe.NAHTAVILLAOLO, VaiheenTila.ODOTTAA_HYVAKSYNTAA);
+        loadProjektiByOidStub.resolves(projekti);
+      });
+
+      it("should send emails and attachments succesfully", async () => {
+        publishProjektiFileStub.resolves();
+        synchronizeProjektiFilesStub.resolves();
+        updateNahtavillaoloKuulutusJulkaisuStub.resolves();
+        s3Mock.s3Mock.on(GetObjectCommand).resolves({
+          Body: Readable.from(""),
+          ContentType: "application/pdf",
+        } as GetObjectCommandOutput);
+
+        await expect(nahtavillaoloTilaManager.approve(projekti, UserFixture.hassuATunnus1)).to.eventually.be.fulfilled;
+        expectAwsCalls("s3Mock", s3Mock.s3Mock.calls());
+        emailClientStub.verifyEmailsSent();
+      });
+
+      describe("sendNahtavillaoloMails", () => {
+        beforeEach(() => {
+          projekti = new DBProjektiForSpecificVaiheFixture().getProjektiForVaihe(Vaihe.NAHTAVILLAOLO, VaiheenTila.HYVAKSYTTY);
+          assertIsDefined(projekti.nahtavillaoloVaiheJulkaisut?.[0]?.hyvaksymisPaiva);
+
+          const julkaisu1 = projekti.nahtavillaoloVaiheJulkaisut[0];
+          projekti.nahtavillaoloVaiheJulkaisut = [
+            julkaisu1,
+            {
+              ...julkaisu1,
+              aineistoMuokkaus: { alkuperainenHyvaksymisPaiva: julkaisu1.hyvaksymisPaiva! },
+              tila: KuulutusJulkaisuTila.ODOTTAA_HYVAKSYNTAA,
+            },
+          ];
+
+          loadProjektiByOidStub.resolves(projekti);
+        });
+
+        it("dontSendIlmoitusMailsForAineistoMuokkaus", async () => {
+          publishProjektiFileStub.resolves();
+          synchronizeProjektiFilesStub.resolves();
+          updateNahtavillaoloKuulutusJulkaisuStub.resolves();
+          s3Mock.s3Mock.on(GetObjectCommand).resolves({
+            Body: Readable.from(""),
+            ContentType: "application/pdf",
+          } as GetObjectCommandOutput);
+          await expect(nahtavillaoloTilaManager.approve(projekti, UserFixture.pekkaProjari)).to.eventually.be.fulfilled;
+          expectAwsCalls("s3Mock", s3Mock.s3Mock.calls());
+          emailClientStub.verifyEmailsSent();
+        });
```

Comment:
Tässä testaat snapshotilla, että kutsuja ei lähde, mutta olisi parempi, että testit speksaavat, mikä haluttu toiminto on. Snapshotin voi pitää kenties rinnalla, mut olennaisempaa olis että testi spesifisti testaa sitä, mitä it-tekstissä väitetään että se testaa.

Reply:
No mietin samaa. Teen sen niin.